sql/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java

```diff
@@ -687,9 +687,60 @@ public void testAlterShardsOfPartitionedTable() throws Exception {
         assertThat(settingsResponse.getSetting(partition, IndexMetaData.SETTING_NUMBER_OF_SHARDS), is("5"));
     }
 
+    @Test
+    public void testAlterShardsTableCombinedWithOtherSettingsIsInvalid() throws Exception {
+        execute("create table quotes (id integer, quote string, date timestamp) " +
+                "clustered into 3 shards with (number_of_replicas='0-all')");
+
+        expectedException.expect(SQLActionException.class);
+        expectedException.expectMessage("Setting [number_of_shards] cannot be combined with other settings");
+        execute("alter table quotes set (number_of_shards=1, number_of_replicas='1-all')");
+    }
+
+    @Test
+    public void testAlterShardsTableSameNumberOfShardsIsInvalid() throws Exception {
+        execute("create table quotes (id integer, quote string, date timestamp) " +
+                "clustered into 3 shards with (number_of_replicas='0-all')");
+
+        expectedException.expect(SQLActionException.class);
+        expectedException.expectMessage("Table/partition is already allocated <3> shards");
+        execute("alter table quotes set (number_of_shards=3)");
+    }
+
+    @Test
+    public void testAlterShardsTableIncreaseNumberOfShardsIsInvalid() throws Exception {
+        execute("create table quotes (id integer, quote string, date timestamp) " +
+                "clustered into 3 shards with (number_of_replicas='0-all')");
+
+        expectedException.expect(SQLActionException.class);
+        expectedException.expectMessage("Increasing the number of shards is not supported");
+        execute("alter table quotes set (number_of_shards=5)");
+    }
+
+    @Test
+    public void testAlterShardsTableReadWriteTableIsInvalid() throws Exception {
+        execute("create table quotes (id integer, quote string, date timestamp) " +
+                "clustered into 3 shards with (number_of_replicas='0-all', \"blocks.write\"=false)");
+
+        expectedException.expect(SQLActionException.class);
+        expectedException.expectMessage("Table/Partition needs to be at a read-only state");
+        execute("alter table quotes set (number_of_shards=1)");
+    }
+
+    @Test
+    public void testAlterShardsTable() throws Exception {
+        execute("create table quotes (id integer, quote string, date timestamp) " +
+                "clustered into 3 shards with (number_of_replicas='0-all', \"blocks.write\"=true)");
+
+        execute("alter table quotes set (number_of_shards=1)");
+        GetSettingsResponse settingsResponse = client().admin().indices().prepareGetSettings("quotes").execute().get();
+        assertThat(settingsResponse.getSetting("quotes", IndexMetaData.SETTING_NUMBER_OF_SHARDS), is("1"));
```

Comment:
why are you using the es api here? the sql below is checking the same...

Reply:
removed es api